frcxx_nft/src/lib.rs

```diff
@@ -110,8 +113,8 @@ where
 
     /// Return the number of NFTs held by a particular address
     pub fn balance_of(&self, address: &Address) -> Result<u64> {
-        let balance = match self.msg.resolve_id(address) {
-            Ok(owner) => self.state.get_balance(&self.bs, owner)?,
+        let balance = match self.helper.resolve_id(address) {
+            Ok(owner) => self.state.get_balance(&self.helper.runtime, owner)?,
```

Comment:
The difference between sometimes calling a method on `helper` or sometimes extracting the `runtime` from it suggests something slightly off in this design. Either:
- abstract the runtime with the helper (make it private), forwarding methods, or
- turn the helper into a collection of free functions that take a runtime parameter, no struct; hold an `RT` object in the NFT struct.

The helper struct boils down to just currying in the runtime object to a bunch of helper functions, but I don't think it adds utility. I'd probably go for the second approach, but ok with either.

Then
```
        let balance = match util::resolve_id(self.runtime, address) {
            Ok(owner) => self.state.get_balance(&self.runtime, owner)?,
```

Reply:
When you reviewed, I think the only places where `helper.runtime` was used was in methods that expected an `impl Blockstore` param. 

I've now made the `ActorHelper` itself implement the trait rather than having the runtime do it so the abstraction feels less leaky now